web/src/components/form.tsx

```diff
@@ -0,0 +1,292 @@
+import { API } from '../constants';
+import typography from '../typography';
+import { ShadowBox } from './UI/shadowContainer';
+
+export interface IFormProps {
+    children?: React.ReactNode,
+    onSubmit?: (e: any) => void
+};
+
+const axi = axios.create({
+    baseURL: API.BASE_URL(),
+    // headers: {
+    //     Authorization: `Bearer ${keycloak?.token}`
+    // },
+});
+console.log(`API Base URL = ${API.BASE_URL()}`);
+
+const StyledTitle = styled.h1`
+    ${typography.toString()}
+    font-size: 24px;
+    font-weight: bold;
+    font-stretch: normal;
+    font-style: normal;
+    line-height: normal;
+    letter-spacing: normal;
+    color: #036;
+`;
+
+const MyForm: React.SFC<IFormProps> = (props) => {
+    // const { keycloak } = useKeycloak();
+    const [state, setState] = useState<any>([]);
+
+    const onSubmit = async (data: any) => {
+        const headers = {
+            // TODO: Add authentication header.
+        };
+        const profile = {
+            name: 'Health Gateway II',
+            description: 'This is a cool website',
+            busOrgId: 'HLTH',
+            prioritySystem: false
+        };
+        const pcontact = {
+            firstName: 'Jason',
+            lastName: 'Leach',
+            email: 'jason.leach@fullboar.ca',
+            githubId: 'jleach',
+            roleId: 1,
+        }
+        const tcontact = {
+            firstName: 'Phill',
+            lastName: 'Billips',
+            email: 'phill.billips@fullboar.ca',
+            githubId: 'githubPB',
+            roleId: 2,
+        }
+
+        try {
+            // 1. Create the project profile.
+            const response: any = await axi.post('profile', profile, headers);
+            const profileId = response.data.id;
+
+            // 2. Create people and relate to project profile.
+            const x: any = await axi.post('contact', pcontact, headers);
+            const y: any = await axi.post('contact', tcontact, headers);
+
+            // 3. Link the contacts to the profile.
+            await axi.post(`profile/${profileId}/contact/${x.data.id}`, headers);
+            await axi.post(`profile/${profileId}/contact/${y.data.id}`, headers);
+
+            // 4. Create the provisioning request (submit clusters);
+            // const provision = {
+            //     profileId: response.id,
+            //     clusters: [1, 3],
+            // }
+            // await axi.post('provision', provision, headers);
+
+            // 5. All good? Tell the user.
+        } catch (err) {
+            console.log(err);
+        }
+    };
+
+    const validate = (values: any): any => {
+        console.log('validate = ', values);
+    };
+
+    // if (keycloak && keycloak.authenticated) {
+    //     console.log('XXXXXXXXXXXX')
+    //     axi.defaults.headers.common['Authorization'] = `Bearer ${keycloak?.token}`;
+    // }
+
+    useEffect(() => {
+        async function wrap() {
+            const result = await axi.get('ministry', {
```

Comment:
a good pattern is to setstate for a loading indicator as well as errors so that you know when they happen. Now if theres like a global loading indicator you can set it using some type of higher order component. This is how global snackbars work